cmd/user/internal/interfaces/http/handlers/auth.go

```diff
@@ -2,41 +2,67 @@ package handlers
 
 import (
 	"encoding/json"
+	"log"
 	"net/http"
+	"time"
 
+	"github.com/dgrijalva/jwt-go"
+	"github.com/google/uuid"
+	"github.com/markbates/goth"
 	"github.com/markbates/goth/gothic"
 	"github.com/vardius/go-api-boilerplate/cmd/user/internal/domain/user"
 	"github.com/vardius/go-api-boilerplate/internal/commandbus"
 	"github.com/vardius/go-api-boilerplate/internal/errors"
 	"github.com/vardius/go-api-boilerplate/internal/http/response"
 )
 
-type requestBody struct {
-	Email string `json:"email"`
+// User contains the information common amongst most OAuth and OAuth2 providers.
+// All of the "raw" datafrom the provider can be found in the `RawData` field.
+type User struct {
```

Comment:
how does this work ? can you provide example for me of curl request how it would look like ?
so i can understand better

Reply:
When we create a jwt token without an oauth2 server all we need is to decide what our Claims struct will be. The internal one is based on internal user id  so we keep it same way for the social auth registration. In order for that to work we need to generate the id in the BuildSocialAuthHandler handler, store it in the User struct claims and pass it to the RegisterWithProvider command to reuse it.